app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCard.kt

```diff
@@ -0,0 +1,74 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.mock.MoviesMock
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.tooling.preview.Preview
+import coil.compose.AsyncImage
+
+data class MovieCardViewState(
+    val imageUrl: String,
+    var isFavorite: Boolean
+)
+
+@Composable
+fun MovieCard(
+    movieCardViewState: MovieCardViewState,
+    modifier: Modifier = Modifier,
+    onFavouriteClicked: () -> Unit,
+    onMovieCardClicked: () -> Unit
+) {
+    Card(
+        modifier = modifier.clickable { onMovieCardClicked() },
+        shape = RoundedCornerShape(dimensionResource(id = R.dimen.rounded_corner_shape))
+    ) {
+        Box(
+            modifier = modifier.fillMaxSize(), contentAlignment = Alignment.TopStart
+        ) {
+            AsyncImage(
+                model = movieCardViewState.imageUrl,
+                contentDescription = null,
+                contentScale = ContentScale.Crop,
+                modifier = modifier.fillMaxSize()
+            )
+            FavouriteButton(color = colorResource(R.color.black_favourite_btn).copy(ButtonConstant.BUTTON_TRANSPARENCY),
+                modifier = modifier
```

Comment:
`modifier = Modifier.padding(...)`
Again, we wan't to create a new modifier here, not reuse.